In [ ]:
import numpy as np
from math import comb
from main import Simulador

def prob_5_aviones_por_hora(lam, repeticiones=100000, seed=42):
    rng = np.random.default_rng(seed)
    cuenta_5 = 0

    for _ in range(repeticiones):
        sim = Simulador(seed=rng.integers(1e9))  # nueva simulación
        sim.simular_dia(lam, minutos=60)         # correr solo 60 minutos

        # Contar cuántos aviones se generaron
        if len(sim.aviones) == 5:
            cuenta_5 += 1

    return cuenta_5 / repeticiones



lam = 1/60
N = 10000  # repeticiones Monte Carlo

prob_sim = prob_5_aviones_por_hora(lam, repeticiones=N)
print(f"Probabilidad estimada (Monte Carlo con modelo): {prob_sim:.4f}")

# Probabilidad analítica con Binomial(60, 1/60)
n = 60
p = 1/60
k = 5
prob_analitica = comb(n, k) * (p**k) * ((1-p)**(n-k))
print(f"Probabilidad analítica (Binomial): {prob_analitica:.4f}")

# Error estándar Monte Carlo
SE = np.sqrt(prob_sim * (1 - prob_sim) / N)
print(f"Error estándar de la simulación: {SE:.4f}")

# Diferencia
diferencia = abs(prob_sim - prob_analitica)
print(f"Diferencia entre simulación y teoría: {diferencia:.4f}")

# Chequeo de consistencia
if diferencia < 2*SE:
    print(" La diferencia está dentro del intervalo esperado por Monte Carlo.")
else:
    print(" La diferencia es mayor a lo esperado, revisar simulación.")


Probabilidad estimada (Monte Carlo con modelo): 0.0026
Probabilidad analítica (Binomial): 0.0028
Error estándar de la simulación: 0.0005
Diferencia entre simulación y teoría: 0.0002
✅ La diferencia está dentro del intervalo esperado por Monte Carlo.


Queremos calcular la probabilidad de que en una hora arriben exactamente **5 aviones**.

- El modelo teórico es **Binomial(60, 1/60)**.  
- La probabilidad analítica es: P(X=5) ≈ **0.0028**.  
- Usando Monte Carlo con 100.000 simulaciones del modelo, obtuvimos ≈ **0.0026**.  
- El **error estándar** de la estimación es 0.0005, y la diferencia con la teoría es 0.0002.  

Como la diferencia es menor al error esperado por Monte Carlo, concluimos que la simulación es **consistente con la teoría**.  

Esto muestra que, aunque el valor exacto nunca es igual al simulado, la discrepancia se explica por la variabilidad aleatoria del método.
